File: java/rich_game_mechnism/test/com/tw/core/places/PlayerRollToHospitalTest.java
package com.tw.core.places;

import com.tw.core.Dice;
import com.tw.core.Game;
import com.tw.core.GameMap;
import com.tw.core.Player;
import com.tw.core.commands.Command;
import com.tw.core.commands.CommandFactory;
import com.tw.core.commands.Roll;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Matchers.anyObject;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * Created by pzzheng on 11/17/16.
 */
public class PlayerRollToHospitalTest {
    public static final int INITIAL_FUND = 10000;
    private GameMap map;
    private Dice dice;
    private Hospital hospital;

    @Before
    public void setUp() {
        map = mock(GameMap.class);
        dice = mock(Dice.class);
        hospital = new Hospital();
    }

    @Test
    public void should_get_into_prison_for_2_days_and_end_turn() {
        when(map.move(anyObject(), anyInt())).thenReturn(hospital);
        Game game = mock(Game.class);
        when(game.getMap()).thenReturn(map);
        Player player = Player.createPlayerWithGame_Fund_CommandState(game, INITIAL_FUND);

        assertThat(player.getStatus(), is(Player.Status.WAIT_FOR_COMMAND));

        Command roll = CommandFactory.Roll(dice);
        player.execute(roll);

        assertThat(player.getStatus(), is(Player.Status.WAIT_FOR_TURN));
        assertThat(player.getCurrentPlace(), is(hospital));
        assertThat(player.isStucked(), is(false));
        assertThat(player.lastCommand() instanceof Roll, is(true));

    }
}

File: ruby/rich_game_ruby/test/player_roll_to_hospital_test.rb
require 'minitest/autorun'
require 'minitest/mock'
require '../src/commands/command_factory'
require '../src/player'
require '../src/places/hospital'
require '../src/assistencePower/tool'

class PlayerRollToHospitalTest < Minitest::Test
  def setup
    @hospital = Hospital.new
    @map = Minitest::Mock.new
    @map.expect(:move, @hospital, [Object, Fixnum])
    @game = Minitest::Mock.new(@map)
    @game.expect(:map, @map)
    @game.expect(:nextPlayer, [])
    @dice = Minitest::Mock.new.expect(:next, 1)
    @rollCommand = CommandFactory.Roll(@dice)
  end

  def test_that_end_turn
    player = Player::create_player_with_game_and_fund_and_command_state(@game)

    player.execute(@rollCommand)

    assert_equal @hospital, player.currentPlace
    assert_equal player.status, Player::Status::WAIT_FOR_TURN
    refute player.isStuck
  end

end